components/eamxx/src/share/field/field_layout.cpp

```diff
@@ -158,17 +158,16 @@ LayoutType get_layout_type (const std::vector<FieldTag>& field_tags) {
       // Possible supported scenarios:
       //  1) <CMP|TL,LEV|ILEV>
       //  2) <TL,CMP>
-      if ( is_lev_tag(tags[1]) and is_vec_tag(tags[0]) ) {
+      if ( (is_vec_tag(tags[0]) or tags[0]==TL) and is_lev_tag(tags[1]) ) {
         result = LayoutType::Vector3D;
-      } else if (tags[0]==TL && tags[1]==CMP ) {
+      } else if (tags[0]==TL && is_vec_tag(tags[1]) ) {
         result = LayoutType::Tensor2D;
       }
       break;
     case 3:
       // The only supported scenario is:
       //  1) <TL,  CMP, LEV|ILEV>
```

Comment:
I guess we could update the comments in this section of code to not say CMP but something including all possible vector tags.